pkg/collector/py/loader.go

```diff
@@ -160,7 +188,45 @@ func (cl *PythonCheckLoader) Load(config integration.Config) ([]check.Check, err
 			} else {
 				log.Debugf("python check '%s' doesn't have a '__version__' attribute: %s", config.Name, errors.New(pyErr))
 			}
-			log.Infof("python check '%s' doesn't have a '__version__' attribute", config.Name)
+			log.Debugf("python check '%s' doesn't have a '__version__' attribute", config.Name)
+
+		}
+
+		if !agentConfig.Datadog.GetBool("disable_py3_validation") && !loadedAsWheel {
+			// Customers, though unlikely might version their custom checks.
+			// Let's use the module namespace to try to decide if this was a
+			// custom check, check for py3 compatibility
+			checkFilePath := checkModule.GetAttrString("__file__")
+
+			tags := []string{"check_name:" + name, "agent_version:6"}
+			if checkFilePath != nil {
+				defer checkFilePath.DecRef()
+				if python.PyString_Check(checkFilePath) {
+					filePath := python.PyString_AsString(checkFilePath)
+
+					// __file__ return the .pyc file path
+					if strings.HasSuffix(moduleName, ".pyc") {
+						filePath = filePath[:len(filePath)-1]
+					}
+					if warnings, err := validatePython3(name, filePath); err == nil {
+						val, status := addExpvarPy3Warnings(name, warnings)
+						tags = append(tags, fmt.Sprintf("status:%s", status))
+						cl.telemetry.Gauge(a7ReadyMetric, val, "", tags)
+					} else {
+						tags = append(tags, fmt.Sprintf("status:%s", a7TagUnknown))
+						cl.telemetry.Gauge(a7ReadyMetric, a7NotReady, "", tags)
+						log.Warnf("Could not lint check %s for Python3 compatibility: %s", name, err)
+					}
+				} else {
+					tags = append(tags, fmt.Sprintf("status:%s", a7TagUnknown))
+					cl.telemetry.Gauge(a7ReadyMetric, a7NotReady, "", tags)
```

Comment:
I think we should not send the value `a7NotReady` when the status is unknown, even if we do set an "unknown" tag. I think it'd be too easy to confuse it with the "not ready" case.

We may want to send a different value altogether when the validation itself failed to complete though

Reply:
I added the `unknown` tag. I think it's safe to say that if the linter failed on that check it's not going to run with python3. Wdyt ?

Reply:
I think if the linter fails we should assume the worst and blame the python module, +1 for Maxime's solution

Reply:
ok, works for me 👍